### PR TITLE
Add `oneOf` support

### LINKED ISSUE #57: C# `oneOf` support
Polymorphic types should be supported via the `oneOf` type.

This should probably be dependent on #55.

### PR DESCRIPTION
**Pull request by mdekrey**
Adds `oneOf` support for C# and TypeScript codegeneration.

Resolves #57

### CODE REVIEW

File: lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿using HandlebarsDotNet;
 using Microsoft.OpenApi.Models;
+using PrincipleStudios.OpenApi.CSharp.templates;
```
Comment (Mike343) [inline lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs:3]: Is `templates` lowercased on purpose?

File: lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+public class OneOfYamlShould
+{
+    private readonly CommonDirectoryFixture commonDirectory;
+
+    public OneOfYamlShould(CommonDirectoryFixture commonDirectory)
+    {
+        this.commonDirectory = commonDirectory;
+    }
+
+
```
Comment (Mike343) [inline lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs:21]: Nitpick, extra line break.

File: lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    [Theory]
+    [MemberData(nameof(ModelsWithoutDiscriminator))]
+    public async Task Allow_a_model_without_discriminator(object body)
+    {
+        var result = await commonDirectory.CheckModel("one-of.yaml", "Pet", body);
+
+        Assert.Equal(0, result.ExitCode);
+    }
+    public static IEnumerable<object[]> ModelsWithoutDiscriminator()
```
Comment (Mike343) [inline lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs:30]: Nitpick, missing line break.

File: lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    [Theory]
+    [MemberData(nameof(ModelsWithDiscriminator))]
+    public async Task Allow_a_model_with_a_discriminator(object body)
+    {
+        var result = await commonDirectory.CheckModel("one-of.yaml", "SpecifiedPet", body);
+
+        Assert.Equal(0, result.ExitCode);
+    }
+    public static IEnumerable<object[]> ModelsWithDiscriminator()
```
Comment (Mike343) [inline lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs:44]: Missing line break.

**Review by Mike343 — APPROVED** [review]

File: lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs (hunk 3 above)
Comment (mdekrey) [reply lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs:30]: This missing line break is intentional - the function is tied directly to the `MemberData` attribute above, so this is essentially part of the test above it.

File: lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs (hunk 4 above)
Comment (mdekrey) [reply lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Integration/OneOfYamlShould.cs:44]: This missing line break is intentional - the function is tied directly to the `MemberData` attribute above, so this is essentially part of the test above it.

File: lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs (hunk 1 above)
Comment (mdekrey) [reply lib/PrincipleStudios.OpenApi.CSharp/CSharpSchemaSourceResolver.cs:3]: Legacy reasons here, unfortunately; I'd like to correct a number of these in the future. (Definitely not for this PR.)

**Pull request merged**